.github/workflows/odh-notebooks-sync.yml

```diff
@@ -0,0 +1,132 @@
+# The aim of this GitHub workflow is to update the pipfile to sync with Codeflare-SDK release.
+name: Sync ODH-notebooks with codeflare-sdk release
+on:
+  workflow_dispatch:
+    inputs:
+      upstream-repository-organization:
+        required: true
+        description: "Owner of target upstream notebooks repository used to open a PR against"
+        default: "opendatahub-io"
+
+      codeflare-repository-organization:
+        required: true
+        description: "Owner of origin notebooks repository used to open a PR"
+        default: "project-codeflare"
+
+      codeflare_sdk_release_version:
+        required: true
+        description: "Provide version of the Codeflare-SDK release"
+
+env:
+  BRANCH_NAME: main
+  CODEFLARE_RELEASE_VERSION: ${{ github.event.inputs.codeflare_sdk_release_version }}
+  UPDATER_BRANCH: odh-sync-updater-${{ github.run_id }}
+  UPSTREAM_OWNER: ${{ github.event.inputs.upstream-repository-organization }}
+  REPO_OWNER: ${{ github.event.inputs.codeflare-repository-organization }}
+  REPO_NAME: notebooks
+  GITHUB_TOKEN: ${{ secrets.CODEFLARE_MACHINE_ACCOUNT_TOKEN }}
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ env.BRANCH_NAME }}
```

Comment:
Makes me think, what is SDK repository used for in this workflow?

Reply:
Yes right, I think this step is not required ~

As we are clonning project-codeflare's/notebooks branch freshly after checkout step and again creating new branch using checkout to make changes? Yeah it doesn't makes sense
For example : 
Here by default I'm using `project-codeflare/notebooks` as a base to make changes and create PR against `ODH/notebooks` in this workflow steps

Thank you @sutaakar for pointing this out!
Let me test it again by removing this checkout step!